pkg/libinit/proc_linux.go

```diff
@@ -62,6 +96,57 @@ func linuxDefault(c *exec.Cmd) {
 	}
 }
 
+func Raw(r *os.File) error {
+	termios, err := unix.IoctlGetTermios(int(r.Fd()), unix.TCGETS)
+	if err != nil {
+		return err
+	}
+
+	termios.Iflag &^= unix.IGNBRK | unix.BRKINT | unix.PARMRK | unix.ISTRIP | unix.INLCR | unix.IGNCR | unix.ICRNL | unix.IXON
+	termios.Oflag &^= unix.OPOST
+	termios.Lflag &^= unix.ECHO | unix.ECHONL | unix.ICANON | unix.ISIG | unix.IEXTEN
+	termios.Cflag &^= unix.CSIZE | unix.PARENB
+	termios.Cflag |= unix.CS8
+	termios.Cc[unix.VMIN] = 1
+	termios.Cc[unix.VTIME] = 0
+
+	if err = unix.IoctlSetTermios(int(r.Fd()), unix.TCSETS, termios); err != nil {
+		return err
+	}
+	if err = syscall.SetNonblock(int(r.Fd()), true); err != nil {
+		return err
+	}
+	return nil
+}
+
+func NewPTMS() (*os.File, *os.File, error) {
+	p, err := os.OpenFile("/dev/ptmx", os.O_RDWR, 0)
+	if err != nil {
+		time.Sleep(1 * time.Second)
+		return nil, nil, err
+	}
+
+	// unlock
+	var u int32
+	// use TIOCSPTLCK with a pointer to zero to clear the lock.
+	err = ioctl(p, syscall.TIOCSPTLCK, uintptr(unsafe.Pointer(&u))) //nolint:gosec // Expected unsafe pointer for Syscall call.
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sname, err := ptsname(p)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	t, err := os.OpenFile(sname, os.O_RDWR|syscall.O_NOCTTY|syscall.O_NONBLOCK, 0o620) //nolint:gosec // Expected Open from a variable.
```

Comment:
I don't understand the need for no lint here, what was the issue? We should really try to minimize the use of no lint.

Reply:
We don't use gosec, but maybe we should.